server/src/main/java/org/elasticsearch/script/field/WriteField.java

```diff
@@ -51,7 +53,30 @@ public String getName() {
      * Does the path exist?
      */
     public boolean exists() {
-        return leaf != null && container.containsKey(leaf);
+        return exists(false);
+    }
+
+    public boolean exists(boolean failOutOfRange) {
+        if (leaf == null) {
+            return false;
+        }
+        if (container instanceof Map<?, ?> mapContainer) {
+            return mapContainer.containsKey(leaf);
+        } else if (container instanceof List<?> listContainer) {
+            int i = parseIndex(leaf);
```

Comment:
The resolution method, `resolveDepthFlat` should handle all the resolution.  We shouldn't need to do resolution at call time outside of that method.